src/FbTk/Menu.cc

```diff
@@ -320,15 +320,24 @@ void Menu::lower() {
 
 void Menu::cycleItems(bool reverse) {
 
-    if (m_items.empty())
-        return;
-
-    int offset = reverse ? -1 : 1;
     int l = m_items.size();
-    int i = m_active_index;
+    int i;
+    int offset = reverse ? -1 : 1;
     size_t ignore;
 
-    for (i += offset; i != m_active_index; i += offset ) {
+    // check if there is _any_ selectable item
```

Comment:
I guess performance isn't an issue (in that you'll not get a long list of unselectable items that you end up traversing twice) otherwise you could just decrement a copy of l and return once that hits 0 (as you've now tested all items)

Reply:
the original suggestions of sebastian was a 2 liner which i found harder to understand. performance is not really an issue imho, i cant imagine a menu with 1k entries … and parsing that list twice is … fast as well.

i also considered to rewrite the current code to return a list of indizes of selectable-items, then reverse the whole list if required, and then just cycle. this would transform the current code with the if-else part for the direction in something very straightforward. but … i decided against atm :)